libs/s25main/EconomyModeHandler.h

```diff
@@ -14,49 +15,58 @@ class GameEvent;
 class EconomyModeHandler : public GameObject
 {
 public:
-    static const unsigned int numGoodTypesToCollect = 7;
-
-    struct econTeam
+    // Object to hold data on the teams competing in the economy mode
+    struct EconTeam
     {
-        unsigned mask;
-        unsigned teamAmounts[numGoodTypesToCollect];
-        unsigned num_players_in_team;
-        unsigned teamWins;
-
-        econTeam(unsigned mask, unsigned num_players_in_team) noexcept
-            : mask(mask), num_players_in_team(num_players_in_team)
-        {
-            for(unsigned int& teamAmount : teamAmounts)
-            {
-                teamAmount = 0;
-            }
-            teamWins = 0;
-        }
+        // The players that are in the team
+        std::bitset<MAX_PLAYERS> playersInTeam;
+
+        // The amounts the team collected for each good
+        std::vector<unsigned> amountsTheTeamCollected;
+
+        // The number of good types the team is the leader in
+        unsigned goodTypeWins;
+
+        EconTeam(std::bitset<MAX_PLAYERS> playersInTeam, unsigned int numGoodTypesToCollect) noexcept
+            : playersInTeam(playersInTeam), amountsTheTeamCollected(numGoodTypesToCollect, 0), goodTypeWins(0)
+        {}
+
+        EconTeam(SerializedGameData& sgd, unsigned int numGoodTypesToCollect);
+
+        void Serialize(SerializedGameData& sgd) const;
 
+        // Returns true if the player is in the team
         bool inTeam(unsigned int playerId) const;
```

Comment:
Nit: For such query like function adding verbs is a good idea. Here: `isInTeam` or better:  `containsPlayer`, then it really reads like english on the call site: `team.containsPlayer(playerId)`
You don't need to change it, just a note